UPGRADE-2.8.md

```diff
@@ -42,6 +42,101 @@ Form
        private $author;
    }
    ```
+   
+ * Type names were deprecated and will be removed in Symfony 3.0. Instead of
+   referencing types by name, you should reference them by their
+   fully-qualified class name (FQCN) instead. With PHP 5.5 or later, you should
+   use the "class" constant for that:
+   
+   Before:
+   
+   ```php
+   $form = $this->createFormBuilder()
+       ->add('name', 'text')
+       ->add('age', 'integer')
+       ->getForm();
+   ```
+   
+   After:
+   
+   ```php
+   use Symfony\Component\Form\Extension\Core\Type\FormType;
+   use Symfony\Component\Form\Extension\Core\Type\IntegerType;
+   use Symfony\Component\Form\Extension\Core\Type\TextType;
+   
+   $form = $this->createFormBuilder()
+       ->add('name', TextType::class)
+       ->add('age', IntegerType::class)
+       ->getForm();
+   ```
+   
+   As a further consequence, the method `FormTypeInterface::getName()` was
+   deprecated and will be removed in Symfony 3.0. You should remove this method
+   from your form types.
+   
+   If you define your form types in the Dependency Injection configuration, you
+   should further remove the "alias" attribute:
+   
+   Before:
+   
+   ```xml
+   <service id="my.type" class="Vendor\Type\MyType">
+       <tag name="form.type" alias="mytype" />
+   </service>
+   ```
+   
+   After:
+   
+   ```xml
+   <service id="my.type" class="Vendor\Type\MyType">
+       <tag name="form.type" />
+   </service>
+   ```
+   
+ * Returning type instances from `FormTypeInterface::getParent()` is deprecated
+   and will not be supported anymore in Symfony 3.0. Return the fully-qualified
+   class name of the parent type class instead.
+   
+   Before:
+   
+   ```php
+   class MyType
+   {
+       public function getParent()
+       {
+           return new ParentType();
```

Comment:
ah sorry, I misread the deprecation message. I confused with the deprecation of type names (which is the main goal of this PR even though this other change went in)

Returning an instance in ``getParent`` was indeed a bad idea by forcing to resolve types again and again.